crates/dbsp/src/operator/input.rs

```diff
@@ -1049,16 +1049,15 @@ mod test {
     ) -> AnyResult<InputHandle<OrdZSet<usize, isize>>> {
         let (stream, handle) = circuit.add_input_stream::<OrdZSet<usize, isize>>();
 
-        let mut expected_batches = input_batches()
-            .into_iter()
-            .chain(input_batches().into_iter())
-            .chain(input_batches().into_iter().map(move |batch| {
+        let mut expected_batches = input_batches().into_iter().chain(input_batches()).chain(
+            input_batches().into_iter().map(move |batch| {
                 let mut result = batch.clone();
                 for _ in 1..nworkers {
                     result += batch.clone();
                 }
                 result
```

Comment:
I think this could be significantly optimized by multiplying the weights in `batch` by `nworkers`

Reply:
done